### PR TITLE
Updated support for EIP1577

### LINKED ISSUE #6688: Update ENS resolving implementation to match what is defined in EIP-1577
**What problem are you trying to solve?**

To try and find a better system @arachnid and I created [EIP-1577](https://github.com/ethereum/EIPs/pull/1577) that allows for a better representation of network and content addresses.

**Describe the solution you'd like**

Implement resolution using the new `multiaddr` field when available.

*Issue was copied over from the [Metamask](https://github.com/MetaMask/metamask-extension/issues/5742) repository because of laziness*

### PR DESCRIPTION
**Pull request by jeluard**
fixes #6688 

### Summary:

Updated implementation of EIP1577 to latest version of the specs.

_TESTERS_ Can't be easily tested as AFAIK no websites following this standard are available.

status: ready

### CODE REVIEW

File: src/status_im/browser/core.cljs [hunk 1]
```diff
@@ -90,8 +90,10 @@
 
 (defn resolve-ens-contenthash-callback [hex]
   (let [proto-code (subs hex 2 4)
-        hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 8))))]
-    (if (and (#{"00" "01"} proto-code) hash (not= hash resolver/default-hash))
+        hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 12))))]
+    ;; We only support IPFS
+    ;; TODO Once more implementations / providers are published this will have to be improved
+    (if (and (= "e3" proto-code) hash (not= hash resolver/default-hash))
       (re-frame/dispatch [:browser.callback/resolve-ens-multihash-success proto-code hash])
```
Comment (flexsurfer) [inline src/status_im/browser/core.cljs:100]: sorry for confusion, we support both . ipfs and swarm, so changes needed in resolve-ens-multihash-success, also in resolve-ens-content-callback code needs to be changed, maybe better to move it in constants

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/browser/core.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/browser/core.cljs:100]: How do we support swarm? Do we have a way to access files on swarm?

Comment (flexsurfer) [reply src/status_im/browser/core.cljs:100]: yes, there is swarm-gateways.net/bzz:/

**Comment by jeluard** [comment]
@flexsurfer Updated based on your comments!

File: src/status_im/browser/core.cljs [hunk 2]
```diff
@@ -82,16 +82,20 @@
...
       (re-frame/dispatch [:browser.callback/resolve-ens-contenthash]))))
 
 (defn resolve-ens-contenthash-callback [hex]
   (let [proto-code (subs hex 2 4)
-        hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 8))))]
-    (if (and (#{"00" "01"} proto-code) hash (not= hash resolver/default-hash))
+        hash (when hex (multihash/base58 (multihash/create :sha2-256 (subs hex 12))))]
+    ;; We only support IPFS
+    ;; TODO Once more implementations / providers are published this will have to be improved
+    (if (and ((#{"00" ipfs-proto-code} proto-code) hash (not= hash resolver/default-hash)))
```
Comment (flexsurfer) [inline src/status_im/browser/core.cljs]: are you sure swarm code is 00? looks like it should be e4. also it need to be changed in `resolve-ens-multihash-success`

**Comment by jeluard** [comment]
@flexsurfer Nice catch! Fixed now

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/browser/core.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/browser/core.cljs]: Fixed!

**Review by vitvly — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by asemiankevich** [comment]
i dont know how to check it properly because i dont know any dapp supporting the EIP. I have checked that both our dapps are working as it was before, I mean voting dapp and ENS names dapp. Also e2e are passed. I think we are good to go.

**Pull request merged**